File: 3471-minimum-average-of-smallest-and-largest-elements/3471-minimum-average-of-smallest-and-largest-elements.java
class Solution {
    public double minimumAverage(int[] nums) {
        Arrays.sort(nums);
        double min_average = Double.MAX_VALUE;
        int n = nums.length;

        int l = 0;
        int r = n - 1;

        while(l <= r) {
            double average = ((double)nums[l] + nums[r])/ 2;
            min_average = Math.min(min_average, average);
            l++;
            r--;
        }

        return min_average;
    }
}

File: 3471-minimum-average-of-smallest-and-largest-elements/3471-minimum-average-of-smallest-and-largest-elements.py
class Solution:
    def minimumAverage(self, nums: List[int]) -> float:
        nums.sort()
        n = len(nums)
        l = 0
        r = n - 1
        min_average = float('inf')
        while l <= r:
            average = (nums[l] + nums[r]) / 2
            min_average = min(min_average, average)
            l += 1
            r -= 1

        return min_average
            
        